### DISCUSSION TITLE
CGNAT

### DISCUSSION THREAD

**Discussion by MSotoudeh** [b-m-f/WirtBot#274] [Q&A] [answered]
Can WirtBot make a link over the CG-NAT network like some 4g-lte ISPs?
Does WirtBot use some technique like "UDP hole punching" to solve the problem?

**Comment by b-m-f** [comment, marked as answer]
So WirtBot is installed on a machine with a publicly routed IP adress.

The clients, for example the device behind the NAT will connect to the WirtBot and keep the connection alive with the WireGuard `PersistentKeepalive` setting.
This will ping the WirtBot every 25 seconds.

Traffic also always flows through the WirtBot.

This should keep the connections available.

In case you serve content from a NAT connection like that I think you might want to set it up and test it yourself.
